[wallet/mobile] feat: add whitelist/blacklist to address book

OlegMakarenko opened the pull request:
## Problem
- Blacklist feature is missing but is implemented in Desktop Wallet.
## Solution
- Added "whitelist / blacklist" tabs to contact list
- Added option to select list while creating or editing contact. 
- Make the name field optional for blacklisted contact.
- Updated `Contact` component to display blacklisted contacts.
- Replace address with contact name in transaction list and transaction-graphic.
- Transactions from blocked signers are hidden and it is not possible to cosign them.
- Added filter to show blocked transactions.
- Created reusable component `Tabs` (used in `AddressBook` and `AggregateTransactionDetails`).
- Moved hardcoded text to localization files.
- Updated translations.

### Tests
**Increased coverage from 1.5% to 19.5%**
- Setup tests with `testing-library` and `jest`. Changed versions of the conflicting deps.
- Added mocks for native dependencies. Added mocks and helpers for `react-native-svg`, `store` and `secureStorage`.
- Added unit tests for components:
  - `AddressDisplay`
  - `ContactItem`
  - `Tabs`
  - `TransactionGraphic` (`AccountIcon`)
  - `Trunc`
- Added unit tests for screens:
  - `AddContact`
  - `AddressBook`
  - `ContactProfile`
  - `History`



![image](https://user-images.githubusercontent.com/33131259/188713534-fca955ee-c8df-4167-a8ed-a28f67a841e5.png)

cryptoBeliever commented:
@OlegMakarenko what could be improved:

1. When we add a new contact (Open address book -> "Add Contact") always whitelist is selected by default in the selector. Also when we clicked "Add Contact" from "Black List" tab. I think it may be unexpected for users and maybe better in such a situation to select "Black List" on the list automatically. Or add two options "Add" and "Block" like on the desktop wallet.

2. Steps:
- Go to history view and copy the transaction signer from one of the transactions.
- Go to the address book and black list this address
- Go back to the History view
Result: Transaction with blocked signer has the red text "Blocked" which is fine. But it will be hidden only after we refresh the view. 

3. For blocked contacts we have no option to add a name. I think it should be possible and optional as is on the desktop wallet.

OlegMakarenko commented:
Thanks @cryptoBeliever!

1. Added field preselection depending on selected tab (White List or Black List).
2. Added refresh of the transaction list after the address book update.
3. Returned back `name` field for blacklisted contacts. Made this field optional. Updated `TransactionItem` and `TransactionGraphic` components to fit blocked contact name in the row.

gimre-xymcity commented on __tests__/components/Trunc.spec.js:
```diff
@@ -0,0 +1,116 @@
+            const content = '12345678';
+
+            // Act + Assert:
+            const expectedResult = '123456...';
```
can we turn the actual `...` (three dots) into ellipsis? https://en.wikipedia.org/wiki/Ellipsis
(`&hellip;`)

OlegMakarenko replied:
@gimre-xymcity I tried to replace `...` with `&hellip;` but it is rendered as `&hellip;` text. Not sure if it can work on native.

gimre-xymcity replied:
ok, let's leave as is

gimre-xymcity commented on __tests__/screens/ContactProfile.spec.js:
```diff
@@ -0,0 +1,149 @@
+            renderListTypeTest(isBlackListed, expectedTransactionAllowedText);
+        });
+
+        test('renders the transactions are blocked text when isBlackListed is false', () => {
```
s/is false/is true/

OlegMakarenko replied:
Thanks, fixed.
